plugins/module_utils/managed_cluster_addons/application_manager.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.ocmplus.cm.plugins.module_utils.addon_utils import (
+    check_cluster_management_addon_feature,
+    ensure_klusterlet_addon,
+)
+
+
+class application_manager:
```

Comment:
Don't need to be in this PR, but more for discussion. Last scrum I think @TheRealHaoLiu  also mentioned a super class for common behaviour (e.g. managedcluster_addon creation, available check/wait, feature enablement check)? 

currently those features are in addon_utils as they share same logics, by moving into addon super class, I guess it'll be easier for some of addons to diverge (I would imagine just a reimplement of some functions?) 

what do you think about that direction?

Reply:
I agree with your suggestions

Reply:
I have implemented superclass - subclasses